server/lib/pbench/test/conftest.py

```diff
@@ -35,68 +19,19 @@ def setup(request, pytestconfig):
-    pbench_tmp.mkdir(parents=True, exist_ok=True)
-    pbench_logs = srv_pbench / "logs"
-    pbench_logs.mkdir(parents=True, exist_ok=True)
-    pbench_recv_002 = srv_pbench / "pbench-move-results-receive" / "fs-version-002"
-    pbench_recv_002.mkdir(parents=True, exist_ok=True)
-    pbench_recv = srv_pbench / "pbench-move-results-receive" / "fs-version"
-    pbench_recv.mkdir(parents=True, exist_ok=True)
-    opt_pbench = tmp_d / "opt" / "pbench-server"
-    pbench_bin = opt_pbench / "bin"
-    pbench_bin.mkdir(parents=True, exist_ok=True)
-    pbench_cfg = opt_pbench / "lib" / "config"
-    pbench_cfg.mkdir(parents=True, exist_ok=True)
-    pbench_archive = srv_pbench / "archive" / "fs-version-001"
-    pbench_archive.mkdir(parents=True, exist_ok=True)
-
-    # "Install" the default server configuration file.
-    shutil.copyfile(
-        "./server/lib/config/pbench-server-default.cfg",
-        (pbench_cfg / "pbench-server-default.cfg"),
-    )
-    cfg_file = pbench_cfg / "pbench-server.cfg"
-    cfg_file.write_text(server_cfg_tmpl.format(tmp=tmp_d))
-
-    tb_path = "./server/lib/pbench/test/test-fixtures/upload/log.tar.xz"
-
-    pbench_recv_tb = pbench_recv / "tarball-002"
-    pbench_recv_tb.mkdir(parents=True, exist_ok=True)
-    pbench_recv_bad_tb = pbench_recv / "bad_tarball-002"
-    pbench_recv_bad_tb.mkdir(parents=True, exist_ok=True)
-    pbench_recv_badtb_tb = pbench_recv / "mix_tarball-002"
-    pbench_recv_badtb_tb.mkdir(parents=True, exist_ok=True)
-
-    for rec_tb_dir in [pbench_recv_bad_tb, pbench_recv_badtb_tb]:
-        shutil.copyfile((str(tb_path) + ".md5"), (rec_tb_dir / (bad_tarball + ".md5")))
-
-    for rec_tb_dir in [pbench_recv_tb, pbench_recv_badtb_tb]:
-        shutil.copyfile(tb_path, (rec_tb_dir / tarball))
-        shutil.copyfile((str(tb_path) + ".md5"), (rec_tb_dir / (tarball + ".md5")))
-
     pytestconfig.cache.set("TMP", str(tmp_d))
-    pytestconfig.cache.set("_PBENCH_SERVER_CONFIG", str(cfg_file))
-
     request.addfinalizer(lambda: tmp.cleanup())
 
 
 @pytest.fixture
-def valid_config(pytestconfig, monkeypatch):
-    """
-    Mock a pbench-server.cfg configuration as defined above.
-
-    Args:
-        pytestconfig: pytest environmental configuration fixture
-        monkeypatch: testing environment patch fixture
-
-    Returns:
-        a PbenchServerConfig object the test case can use
-    """
-    cfg_file = pytestconfig.cache.get("_PBENCH_SERVER_CONFIG", None)
-    monkeypatch.setenv("_PBENCH_SERVER_CONFIG", cfg_file)
+def test_logger(pytestconfig):
     tmp = Path(pytestconfig.cache.get("TMP", None))
-
-    # It helps to perform the validation of the config file
-    return get_server_config(), tmp
+    logger = logging.getLogger("testing")
+    handler = logging.FileHandler(tmp / "test.log")
+    handler.setLevel(logging.ERROR)
+    logfmt = (
+        "1970-01-01T00:00:42.000000 {levelname} {name}.{module} {funcName} -- {message}"
+    )
+    pf = PbenchFormatter()
```

Comment:
I added this here because I didn't want internal modules to be called and imported somewhere else. If I add it in test-specific module than the problem will be same, I will have to import these internal modules and call them.

Reply:
I still think that putting the guts of `format()` here would be better.  (See my comment above regarding [unit test access limitations](https://github.com/distributed-system-analysis/pbench/pull/2414#discussion_r699537169).)